helixbass/tree-sitter-lint
# Support per-match query callback

In this PR:
- In order to support rule listeners that would really like to "capture multiple different nodes from a single query match", adjust so that per-rule-listener it can either be "individual-capture-oriented" or "match-oriented" (in terms of the "granularity" with which we invoke the rule listener callback and what it receives as arguments to that callback)

To test:
Per the added test if you're using the `rule!()` macro and have a rule listener with multiple (different) `@whatever`'s and make the listener callback take `captures` as its first param, you should be able to access all of those matched capture nodes for each time the overall query matches

Based on `use-rule-languages`
